infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -258,6 +258,23 @@ public synchronized void handleFailureDetectedMsg(CorfuPayloadMsg<FailureDetecto
     }
 
     /**
+     * Handles the heartbeat request.
+     * It accumulates the metrics required to build
+     * and send the response(NodeMetrics).
+     *
+     * @param msg
+     * @param ctx
+     * @param r
+     */
+    @ServerHandler(type = CorfuMsgType.HEARTBEAT_REQUEST)
+    public synchronized void handleHearbeatRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
```

Comment:
why is this synchronized ? I see all the message handlers are synchronized on the instance. Hopefully we have given enough thought to the class members to arrive at the conclusion that all external facing methods should be synchronized.